fast_faults.py

```diff
@@ -136,13 +137,40 @@ def update_filters(self):
             gb = self.findChild(QtWidgets.QGroupBox, f"ff_filter_gb_{opt['name']}")
             cb = self.findChild(QtWidgets.QComboBox, f"ff_filter_cb_{opt['name']}")
             if gb.isChecked():
-                opt['condition'] = cb.currentIndex()
+                opt['condition'] = '"' + str(cb.currentText()).upper() + '"'
                 filters.append(opt)
         self.filters_changed.emit(filters)
 
+    def setup_datetimes(self):
+        self.timer = QtCore.QTimer()
+        self.timer.timeout.connect(self.update_min_times)
+        self.timer.singleShot(1000, self.update_min_times)
+        self.timer.start(10000)
+        self.dt_channel = PyDMChannel(
+            self.ui.PyDMDateTimeLabel.channel,
+            value_slot=self.update_time_delta,
+        )
+        self.dt_channel.connect()
+        self.destroyed.connect(
```

Comment:
Between the destroyed signal and `__dtor__`, I wonder what the real differences are.
I also just stumbled on `WA_DeleteOnClose` which may help us in certain scenarios. Hmm...

Reply:
I don't like having so many options for something so straightforward... "Things to do before deallocating memory" doesn't seem like a thing that needs options